docs/examples/contrib/sqlalchemy/plugins/tutorial/full_app_no_plugins.py

```diff
@@ -64,6 +64,11 @@ async def get_todo_list(done: Optional[bool], session: AsyncSession) -> List[Tod
     return result.scalars().all()
 
 
+async def on_startup(app: Litestar) -> None:
+    async with app.state.engine.begin() as conn:
+        await conn.run_sync(Base.metadata.create_all)
```

Comment:
This would be more succinct if we perform it inside the `db_connection` lifespan context:

```py
@asynccontextmanager
async def db_connection(app: Litestar) -> AsyncGenerator[None, None]:
    engine = getattr(app.state, "engine", None)
    if engine is None:
        engine = create_async_engine("sqlite+aiosqlite:///todo.sqlite", echo=True)
        app.state.engine = engine

    async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.create_all)

    try:
        yield
    finally:
        await engine.dispose()
```

This way we don't need the extra `on_startup` method and it functionality-wise it should be to the same effect.

Reply:
The `asnyccontextcontext` is only there for the first part of the tutorial. For the 2 last parts we need to have the `on_startup` method anyway and I figured I just introduce it there to decrease the amount of changes along the way.

Reply:
Good point.

Got me thinking we should add a `create_all` option to the configs, e.g., `db_config = SQLAlchemyAsyncConfig(connection_string="sqlite+aiosqlite:///todo.sqlite", create_all=metadata)`.

I think use the lifecycle hook in the examples where you can, b/c it is nicer, and switch to using the `on_shutdown` hook when you have to, with an explanation.

I'll add a feature request to `advanced-alchemy` to see if we can get the `create_all` config parameter across the line.

Reply:
https://github.com/jolt-org/advanced-alchemy/issues/65

Reply:
Perfect, thanks!

Reply:
Just updated the first parts of the tutorial. 

@peterschutt and @cofin, is there any reason to limit `advanced-alchemy` to `<0.4.0` in `pyproject.toml`: `sqlalchemy = ["advanced-alchemy>=0.2.2,<0.4.0"]`?

The [aforementioned PR](https://github.com/jolt-org/advanced-alchemy/pull/70) pushed `advanced-alchemy` to v0.4.0. 

For this to work I would limit it to a version below v0.5.0 and then continue?

Reply:
I had to bump the version of `advanced-alchemy` in pdm lock as well to make the CI/CD run. 

LMK if you are happy with the current state.

Reply:
@orgarten We definitely should bump up the advanced alchemy version now.  I'll make a PR to do this today.

Reply:
I did this as part of a PR yesterday to update deps - just capped AA at <1.0